src/components/similarEntries/container/similarEntries.tsx

```
@@ -0,0 +1,153 @@
+    if (!Array.isArray(data) || !post) return [];
+    const seenAuthors = new Set<string>();
+    const out: SearchApiRow[] = [];
+    for (const r of data) {
+      if (!r || typeof r.author !== 'string' || typeof r.permlink !== 'string') continue;
+      if (r.permlink === post.permlink) continue;
+      if ((r.tags ?? []).indexOf('nsfw') !== -1) continue;
+      if (seenAuthors.has(r.author)) continue;
+      seenAuthors.add(r.author);
+      out.push(r);
+      if (out.length >= MAX_RESULTS) break;
+    }
```

ESLint

[error] iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations.

(no-restricted-syntax)